src/rust/engine/src/rule_graph.rs

```diff
@@ -560,7 +608,7 @@ fn function_str(func: &Function) -> String {
 }
 
 
-fn type_str(type_id: TypeId) -> String {
+pub fn type_str(type_id: TypeId) -> String {
```

Comment:
Yeah. I noticed that there are multiple str reps for these things. That could do with consolidation.
I think these don't need to be public anymore, I'll do a pass and rm the pub from the ones that are not used externally. https://github.com/pantsbuild/pants/issues/4377